lib/coverage.js

```diff
@@ -32,8 +32,37 @@ function Coverage(config) {
             return;
         }
 
+        function doesOneRegexMatch(regexes, value) {
+            var results = regexes.map(function(regex) {
+                try {
+                    return regex.test(value);
+                }
+                catch(err) {
+                    throw new Error(value + " is not a regex, please provide only regexes");
+                }
+            });
+
+            var matches = results.filter(function(result) {
+                return result;
+            });
+
+            return matches.length > 0;
+        }
+
+        function checkRegex(regex, value) {
+            if (Array.isArray(regex)) {
+                return doesOneRegexMatch(regex, value);
+            }
+            else if (regex instanceof RegExp) {
+                return regex.test(value);
+            }
+            else {
+                throw new Error("karmaTypescriptConfig.coverageOptions.exclude must be a single RegExp or an Array of RegExp");
```

Comment:
This check should be done in [configuration.js](https://github.com/monounity/karma-typescript/blob/2.1.8/lib/configuration.js#L68), config errors should fail fast before running the preprocessor, compiler etc